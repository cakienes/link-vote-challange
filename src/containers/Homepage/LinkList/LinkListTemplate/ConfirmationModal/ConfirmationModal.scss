@import '../../../../variables.scss';
.confirmationModalHolder {
    .modal-content {
        border-radius: 0%;
        width: 500px;
        display: flex;
        flex-direction: column;
        text-align: center;
        background-color: #d8d8d8;
        .modal-header {
            height: 50px;
            padding: 6px 10px;
            border: none;
            color: #fff;
            background: #000;
            border-radius: 0%;
            button {
                font-size: 40px;
                color: #fff;
                margin: -1.4rem -1rem -1rem auto;
            }
        }

        .question {
            font-size: 22px;
            color: $secondLabelColor;
        }

        .linkName {
            font-size: 35px;
        }
    }

    .modal-footer {
        border: none;
        justify-content: center;
    }
}
