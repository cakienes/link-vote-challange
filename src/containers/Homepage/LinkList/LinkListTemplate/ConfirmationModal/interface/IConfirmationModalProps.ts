import ILink from '../../../../../Interfaces/ILink';

export default interface IConfirmationModalProps {
    data: ILink;
    onClickOk: () => void;
    onClickCancel: () => void;

    visibility?: boolean;
}
