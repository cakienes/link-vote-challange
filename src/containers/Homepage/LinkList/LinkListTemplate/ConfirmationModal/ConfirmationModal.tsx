import React from 'react';
import Modal from 'react-bootstrap/Modal';
import Button from '../../../../../components/Button/Button';
import './ConfirmationModal.scss';
import IConfirmationModalProps from './interface/IConfirmationModalProps';

class ConfirmationModal extends React.Component<IConfirmationModalProps, any> {
    render() {
        const { data } = this.props;
        const { visibility } = this.props;
        return (
            <Modal show={visibility} onHide={this.onClickCancel} className="confirmationModalHolder">
                <Modal.Header closeButton>
                    <Modal.Title>Modal title</Modal.Title>
                </Modal.Header>

                <Modal.Body>
                    <div className="modalContent">
                        <div className="question">Do you want to remove:</div>
                        <div className="linkName">{data.name}</div>
                    </div>
                </Modal.Body>

                <Modal.Footer>
                    <Button label="OK" onClick={this.onClickOk} type="button" />
                    <Button label="CANCEL" onClick={this.onClickCancel} type="button" />
                </Modal.Footer>
            </Modal>
        );
    }
    onClickOk = (): void => {
        this.props.onClickOk();
    };
    onClickCancel = (): void => {
        this.props.onClickCancel();
    };
}

export default ConfirmationModal;
