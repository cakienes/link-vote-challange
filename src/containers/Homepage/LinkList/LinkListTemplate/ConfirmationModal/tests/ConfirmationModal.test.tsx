import { mount } from 'enzyme';
import React from 'react';
import ILink from '../../../../../Interfaces/ILink';
import ConfirmationModal from '../ConfirmationModal';
import IConfirmationModalProps from '../interface/IConfirmationModalProps';

describe('Homepage.tsx', () => {
    const dumLink: ILink = {
        id: 'id',
        point: 1,
        createdAt: new Date(),
        link: 'link',
        name: 'name',
        updatedAt: new Date(),
    };

    const onClickCancelMock = jest.fn();
    const onClickOkMock = jest.fn();

    const confirmationModalProps: IConfirmationModalProps = {
        data: dumLink,
        onClickCancel: onClickCancelMock,
        onClickOk: onClickOkMock,
        visibility: true,
    };

    beforeEach(() => {
        jest.spyOn(global.Date, 'now').mockImplementationOnce(() => new Date('2019-06-02T17:13:58.135Z').valueOf());
    });

    it('renders without crashing', async () => {
        const wrapper = mount(<ConfirmationModal {...confirmationModalProps} />);
        const instance: any = wrapper.instance();

        expect(instance.render()).toMatchSnapshot();
    });

    it('onClickOk of LinkList ', async () => {
        const wrapper = mount(<ConfirmationModal {...confirmationModalProps} />);
        const instance: any = wrapper.instance();

        await instance.onClickOk();
        expect(onClickOkMock).toHaveBeenCalled();
    });

    it('onClickCancel of LinkList ', async () => {
        const wrapper = mount(<ConfirmationModal {...confirmationModalProps} />);
        const instance: any = wrapper.instance();

        await instance.onClickCancel();
        expect(onClickCancelMock).toHaveBeenCalled();
    });
});
