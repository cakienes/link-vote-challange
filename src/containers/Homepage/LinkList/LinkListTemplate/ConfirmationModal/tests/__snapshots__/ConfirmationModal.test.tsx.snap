// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage.tsx renders without crashing 1`] = `
<ForwardRef(render)
  className="confirmationModalHolder"
  onHide={[Function]}
  show={true}
>
  <ForwardRef
    closeButton={true}
    closeLabel="Close"
  >
    <ForwardRef>
      Modal title
    </ForwardRef>
  </ForwardRef>
  <ForwardRef>
    <div
      className="modalContent"
    >
      <div
        className="question"
      >
        Do you want to remove:
      </div>
      <div
        className="linkName"
      >
        name
      </div>
    </div>
  </ForwardRef>
  <ForwardRef>
    <Button
      label="OK"
      onClick={[Function]}
      type="button"
    />
    <Button
      label="CANCEL"
      onClick={[Function]}
      type="button"
    />
  </ForwardRef>
</ForwardRef(render)>
`;
